src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PathSeqFilterSpark.java

```diff
@@ -1,268 +1,115 @@
-        final JavaRDD<GATKRead> readsBQFiltered = readsLengthFiltered.map(new ReadTransformerSparkifier(new BaseQualityReadTransformer(QUAL_PHRED_THRESH)));
-
-        //Filter reads with too many 'N's
-        final JavaRDD<GATKRead> readsAmbigFiltered = readsBQFiltered.filter(new ReadFilterSparkifier(new AmbiguousBaseReadFilter(FRAC_N_THRESHOLD)));
-        logger.info("Reads remaining after ambiguous base filtering: " + readsAmbigFiltered.count());
-
-        //Load Kmer hopscotch set and filter reads containing > 0 matching kmers
-        final JavaRDD<GATKRead> readsKmerFiltered = doKmerFiltering(ctx,readsAmbigFiltered);
-        logger.info("Reads remaining after kmer filtering: " + readsKmerFiltered.count());
-
-        //Filter unpaired reads
-        final JavaRDD<GATKRead> readsFilteredPaired = retainPairs(readsKmerFiltered);
-        logger.info("Reads remaining after unpaired filtering: " + readsFilteredPaired.count());
-
-        //BWA filtering against user-specified host organism reference
-        header = getHeaderForReads();
-        final JavaRDD<GATKRead> readsAligned = doHostBWA(ctx, header, readsFilteredPaired);
-
-        //Get unmapped reads (note these always come in pairs)
-        //TODO: retain read pairs by alignment score instead of flags
-        final JavaRDD<GATKRead> readsNonHost = readsAligned.filter(read -> read.isUnmapped() && read.mateIsUnmapped());
-        logger.info("Reads remaining after BWA filtering: " + readsFilteredPaired.count());
-
-        //TODO: repeat BWA with seed size 11
-
-        //Write output
-        header.setSortOrder(SAMFileHeader.SortOrder.queryname);
-        try {
-            ReadsSparkSink.writeReads(ctx, OUTPUT_PATH, null, readsNonHost, header, shardedOutput ? ReadsWriteFormat.SHARDED : ReadsWriteFormat.SINGLE);
-        } catch (final IOException e) {
-            throw new GATKException("Unable to write bam",e);
+        final SAMFileHeader header = getHeaderForReads();
+        if (filterArgs.alignedInput && (header.getSequenceDictionary() == null || header.getSequenceDictionary().isEmpty())) {
+            logger.warn("--isHostAligned is true but the BAM header contains no sequences");
+        }
+        if (!filterArgs.alignedInput && header.getSequenceDictionary() != null && !header.getSequenceDictionary().isEmpty()) {
+            logger.warn("--isHostAligned is false but there are one or more sequences in the BAM header");
+        }
+        header.setSequenceDictionary(new SAMSequenceDictionary());
+
+        try (final PSFilter filter = new PSFilter(ctx, filterArgs, getReads(), header)) {
+
+            final Tuple2<JavaRDD<GATKRead>, JavaRDD<GATKRead>> result = filter.doFilter();
+            final JavaRDD<GATKRead> pairedReads = result._1;
+            final JavaRDD<GATKRead> unpairedReads = result._2;
+
+            if (!pairedReads.isEmpty()) {
+                final SAMFileHeader.SortOrder oldSortOrder = header.getSortOrder();
```

Comment:
Good point that we don't have to sort by query name here, but the subsequent tool, PathSeqBwaSpark, requires that input read pairs be on the same partition, and the GATK bam reader will only do this if it is query name sorted.

A pipeline tool in a future PR will avoid all this.

I'm not sure what oldSortOrder was for but I'm deleting it.